### ISSUE TITLE
Some jobs have num_retries=-1, even after being run.

### ISSUE THREAD

**Issue by kurtwheeler** [AlexsLemonade/refinebio#1370]
### Context

We used decrementing num_retries as a way to let jobs get requeued in situations where they had to be restarted but it shouldn't have counted against the number of retries for the jobs. For example, when we kill jobs while deploying we want those jobs to be replaced without their retry counter being incremented.

### Problem or idea

I believe that the intention was for jobs to get re-incremented back up to their original count when they got requeued, so we never should have seen jobs actually getting run with num_retries=-1. Also semantically that doesn't really make sense. However I am seeing completed jobs with num_retries=-1:

```
select count(*), num_retries, success from downloader_jobs where created_at > '2019-07-01 00:00:00'::timestamp group by num_retries, success;
 count | num_retries | success 
-------+-------------+---------
  9917 |          -1 | f
 10327 |          -1 | t
  4908 |           0 | f
  8484 |           0 | t
    29 |           0 | 
   921 |           1 | f
  1329 |           1 | t
   686 |           2 | f
   235 |           2 | t
```

### Solution or next step

Figure out why these jobs counter is off and correct it! There's no need to correct this for past jobs, just fix the issue for future jobs.

**Comment by wvauclain** [comment]
I tried running the query again to see how the numbers changed since last time, and I ran into something interesting:

```
select count(*), num_retries, success from downloader_jobs where created_at > '2019-07-01 00:00:00'::timestamp group by num_retries, success;
 count  | num_retries | success
--------+-------------+---------
 118015 |           0 | f
  29661 |           1 | f
  86017 |          -1 | t
  25971 |           2 | f
   3947 |          -1 |
      2 |          -2 | t
  91465 |           0 | t
   2924 |           2 | t
  62286 |          -1 | f
   9661 |           1 | t
     21 |           2 |
     13 |          -2 | f
     67 |           1 |
    644 |           0 |
(14 rows)
```

It looks like there's a couple of downloader jobs that have been retried -2 times.

**Comment by arielsvn** [comment]
@kurtwheeler do you think this might have been fixed by #1726? Seems like there're no jobs with `num_retries=-1` after we deployed that.

```sql
data_refinery=> select count(*), num_retries, success from downloader_jobs where created_at > '2019-10-03 18:40:00'::timestamp group by num_retries, success;
 count | num_retries | success 
-------+-------------+---------
   118 |           0 | f
     4 |           1 | f
 21359 |           0 | t
     1 |           2 | t
   134 |           1 | t
    31 |           0 | 
(6 rows)
```

**Comment by kurtwheeler** [comment]
I gotta say I do think https://github.com/AlexsLemonade/refinebio/pull/1726 fixed this. I was expecting it to once I realized the implications of the bug that PR is fixing, and with those results it looks like it did in fact fix it!

Again, really nice find @arielsvn !!!
